### PR TITLE
Pin JavaScript dependencies

### PR DESCRIPTION
**Pull request by mxdvl**
## What is being recommended?

Recommend pinning dependencies in JavaScript applications. This ensures consistency and reliability.

## What's the context?

https://docs.renovatebot.com/dependency-pinning/#why-pin-dependencies

### CODE REVIEW

**Review by joecowton1 — APPROVED** [review]
Good idea.

**Comment by mxdvl** [comment]
Waiting on #114 to be merged first.

File: dependencies.md [hunk 1]
```diff
@@ -32,12 +32,9 @@ See the [Scala-specific recommendations](./scala.md)
...
+Ensure that all dependencies in deployed applications are pinned to a `patch` version.
+When using a package manager, use a lock file to prevent prevent mismatches in transitive
+dependencies between environments.
 
-- A leading caret `^1.0.0` accepts any subsequent minor or patch version.
-- A leading tilde `~1.1.0` accepts any subsequent patch version.
-
-When developing libraries to be published as NPM packages, [see specific recommendations](./npm-packages.md).
+When developing libraries to be published as NPM packages,
+[use ranges for peer dependencies](./npm-packages.md#peerDependencies).
```
Comment (jamesgorrie) [inline dependencies.md:40]: Might be worth leaving an example of good vs bad, as some people just follow convention and won't know or care what `^` or `~` mean.

Comment (mxdvl) [reply dependencies.md:40]: There are examples in the linked recommendation. Did you have something else in mind?

Comment (jamesgorrie) [reply dependencies.md:40]: in `./npm-packages.md#peerDependencies`? Those don't look like examples, but you could infer that that's the right way I suppose.

**Pull request merged**